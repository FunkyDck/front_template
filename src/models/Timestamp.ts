// ISO8601 string
export type ISOTimestamp = string;
