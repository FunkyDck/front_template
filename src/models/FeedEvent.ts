import { ISOTimestamp } from "./Timestamp"

export type FeedEvent = {
  at: ISOTimestamp
  title: string
  description?: string
}
