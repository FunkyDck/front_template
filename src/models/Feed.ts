import { FeedEvent } from "./FeedEvent"

export type FeedHead = {
  id: string,
  title: string,
}

export type Feed = FeedHead & {
  events: FeedEvent[]
}
