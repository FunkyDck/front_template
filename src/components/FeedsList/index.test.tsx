import { describe, expect, test } from "@jest/globals"
import React from "react"
import renderer from "react-test-renderer"
import { FeedsList } from "."
import { FeedHead } from "../../models/Feed"

const mockFeeds: FeedHead[] = [
  { id: "1", title: "Test 1" },
  { id: "2", title: "Test 2" },
  { id: "3", title: "Test 3" },
]

describe("FeedsList", () => {
  test("Active item is highlighted", () => {
    const component = renderer.create(<FeedsList feeds={mockFeeds} activeId='2' />)

    expect(component.toJSON()).toMatchSnapshot()
  })
})
