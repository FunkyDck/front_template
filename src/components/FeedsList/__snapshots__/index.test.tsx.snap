// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FeedsList Active item is highlighted 1`] = `
<ol
  className="list"
>
  <li
    className=""
    onClick={[Function]}
  >
    Test 1
  </li>
  <li
    className="active"
    onClick={[Function]}
  >
    Test 2
  </li>
  <li
    className=""
    onClick={[Function]}
  >
    Test 3
  </li>
</ol>
`;
